pytorch_lightning/metrics/functional/classification.py

```diff
@@ -173,16 +173,56 @@ def stat_scores_multiple_classes(
     if pred.ndim == target.ndim + 1:
         pred = to_categorical(pred, argmax_dim=argmax_dim)
 
-    num_classes = get_num_classes(pred=pred, target=target,
-                                  num_classes=num_classes)
+    num_classes = get_num_classes(pred=pred, target=target, num_classes=num_classes)
 
-    tps = torch.zeros((num_classes,), device=pred.device)
-    fps = torch.zeros((num_classes,), device=pred.device)
-    tns = torch.zeros((num_classes,), device=pred.device)
-    fns = torch.zeros((num_classes,), device=pred.device)
-    sups = torch.zeros((num_classes,), device=pred.device)
-    for c in range(num_classes):
-        tps[c], fps[c], tns[c], fns[c], sups[c] = stat_scores(pred=pred, target=target, class_index=c)
+    pred = pred.view((-1, )).long()
```

Comment:
is the long (int64) really needed here?

Reply:
Yes. `torch.scatter_add` function requires the 2nd argument (denoting 'index') to be `LongTensor`. If you would like, we can push that type conversion into the `if 'none'` block as the other parts doesn't require it.

Reply:
@Diuven please do that, since this may otherwise also take some time even if not needed

Reply:
OK. Updated accordingly.